typings/index.d.ts

```diff
@@ -1023,7 +1023,7 @@ declare module 'discord.js' {
   export class GuildPreviewEmoji extends BaseGuildEmoji {
     constructor(client: Client, data: object, guild: GuildPreview);
     public guild: GuildPreview;
-    public readonly roles: Set<Snowflake>;
+    public readonly roles: Snowflake[];
```

Comment:
It might be sensible to mark the array as read-only to discourage mutations to it.
```suggestion
    public readonly roles: readonly Snowflake[];
```